### ISSUE TITLE
3.12.x breaks on Java 7 with reflection use

### ISSUE THREAD

**Issue by yschimke** [square/okhttp#6221]
https://gist.github.com/mbjarland/23c28f74ab4c0a786be928891aebfb0a

https://stackoverflow.com/questions/63410786/using-okhttp-3-12-x-from-java-7/63411414?noredirect=1#comment112140880_63411414

cc @mbjarland

**Comment by swankjesse** [comment]
Looks like Groovy does some pretty aggressive introspection. Ugh.

**Comment by yschimke** [comment]
I don't have a viable plan to fix, so mainly I'm flagging for my future self.

**Comment by mbjarland** [comment]
Ok as you guys were talking about introspection I figured it might be worth trying out bypassing the groovy metadata protocol stuff. I added `@CompileStatic` to the method where the issue occurs and voila, problem gone. 

https://docs.groovy-lang.org/latest/html/gapi/groovy/transform/CompileStatic.html

As CompileStatic is applicable both on the method and class level I would say that this fixes the issue. It is in my experience pretty straight forward to make the two-three changes required to make static compilation pass and if that gives me access to this entire lib I'm more than happy with the solution. 

Might be worth adding this to the docs somewhere.
